packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -1,216 +1,460 @@
+        require(
+            _commitment.maxPrincipal > 0,
+            "commitment principal allocation 0"
+        );
+
+        if (_commitment.collateralTokenType != CommitmentCollateralType.NONE) {
+            require(
+                _commitment.maxPrincipalPerCollateralAmount > 0,
+                "commitment collateral ratio 0"
+            );
+
+            if (
+                _commitment.collateralTokenType ==
+                CommitmentCollateralType.ERC20
+            ) {
+                require(
+                    _commitment.collateralTokenId == 0,
+                    "commitment collateral token id must be 0 for ERC20"
+                );
+            }
+        }
+    }
+
     /** External Functions **/
 
     constructor(address _protocolAddress, address _marketRegistry)
         TellerV2MarketForwarder(_protocolAddress, _marketRegistry)
     {}
 
     /**
-     * @notice Updates the commitment of a lender to a market.
-     * @param _tokenAddress The address of the asset being committed.
-     * @param _marketId The Id of the market the commitment applies to.
-     * @param _maxPrincipal Amount of tokens being committed by the lender.
-     * @param _maxLoanDuration Length of time, in seconds that the lender's capital can be lent out for.
-     * @param _minInterestRate Minimum Annual percentage to be applied for loans using the lender's capital.
-     * @param _expiration Expiration time in seconds, when the commitment expires.
+     * @notice Created a loan commitment from a lender to a market.
+     * @param _commitment The new commitment data expressed as a struct
      */
-    function updateCommitment(
-        uint256 _marketId,
-        address _tokenAddress,
-        uint256 _maxPrincipal,
-        uint32 _maxLoanDuration,
-        uint16 _minInterestRate,
-        uint32 _expiration
-    ) public {
-        address lender = _msgSender();
-        require(_expiration > uint32(block.timestamp));
-
-        Commitment storage commitment = lenderMarketCommitments[lender][
-            _marketId
-        ][_tokenAddress];
-        commitment.maxPrincipal = _maxPrincipal;
-        commitment.expiration = _expiration;
-        commitment.maxDuration = _maxLoanDuration;
-        commitment.minInterestRate = _minInterestRate;
-
-        emit UpdatedCommitment(lender, _marketId, _tokenAddress, _maxPrincipal);
+    function createCommitment(
+        Commitment calldata _commitment,
+        address[] calldata borrowerAddressList
+    ) public returns (uint256 commitmentId_) {
+        commitmentId_ = commitmentCount++;
+
+        require(
+            _commitment.lender == _msgSender(),
+            "unauthorized commitment creator"
+        );
+
+        commitments[commitmentId_] = _commitment;
+
+        validateCommitment(commitments[commitmentId_]);
+
+        for (uint256 i = 0; i < borrowerAddressList.length; i++) {
+            commitmentBorrowersList[commitmentId_].add(borrowerAddressList[i]);
+        }
```

Comment:
lets make an `internal` function for this as the same logic is in `updateCommitment`

Reply:
done